### ISSUE TITLE
serverPath not applied to all wasm targets

### ISSUE THREAD

**Issue by jacobmischka** [wasm-tool/rollup-plugin-rust#19]
In my project I have 3 separate wasm targets that I'm including in separate areas of the app. Building works, however, the `serverPath` option is only being applied to one of the three targets, seemingly at random. Inspecting the resulting `bundle.js` shows one of the paths correctly showing `/build/assets/[filename].wasm` in my case, with the other two showing `/assets/[filename].wasm`.

**Comment by jacobmischka** [comment]
As can be seen in my referenced fork commit, removing that state check seems to work around the issue, and doesn't seem to cause any problems in my project at least. I imagine there is some sort of race condition where the URLs are attempting to be resolved not in order with the state changing.

**Comment by Pauan** [comment]
That fix is definitely wrong, since it's adding the `serverPath` to *all* files, not just the `.wasm` files.

That definitely does sound like a race condition. Could you explain more about your setup? For example, could you share your `rollup.config.js`?

**Comment by jacobmischka** [comment]
I thought it would as well, but that function is at least in my case only being called 3 times, once for each wasm entrypoint. I'm not suggesting that it actually be used as a fix or else I would have made a pull request, just as an illustration of where the problem is coming from.

Sure. A lot of the noise in there is development niceties like serving and live reload, but the issue arises when building so none of that weirdness should be relevant.

```js
import svelte from 'rollup-plugin-svelte';
import resolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import css from 'rollup-plugin-css-only';
import livereload from 'rollup-plugin-livereload';
import rust from '@wasm-tool/rollup-plugin-rust';
import { terser } from 'rollup-plugin-terser';

const production = !process.env.ROLLUP_WATCH;

function serve() {
	let server;

	function toExit() {
		if (server) server.kill(0);
	}

	return {
		writeBundle() {
			if (server) return;
			server = require('child_process').spawn('npm', ['run', 'start', '--', '--dev'], {
				stdio: ['ignore', 'inherit', 'inherit'],
				shell: true
			});

			process.on('SIGTERM', toExit);
			process.on('exit', toExit);
		}
	};
}

export default {
	input: 'src/main.js',
	output: {
		sourcemap: true,
		format: 'iife',
		name: 'app',
		file: 'public/build/bundle.js'
	},
	plugins: [
		svelte({
			emitCss: true,
			compilerOptions: {
				// enable run-time checks when not in production
				dev: !production,
			}
		}),

		// If you have external dependencies installed from
		// npm, you'll most likely need these plugins. In
		// some cases you'll need additional configuration -
		// consult the documentation for details:
		// https://github.com/rollup/plugins/tree/master/packages/commonjs
		resolve({
			browser: true,
			dedupe: ['svelte']
		}),
		commonjs(),
		css({
			output: 'bundle.css'
		}),

		rust({
			serverPath: '/build/',
			watchPatterns: ['../../**/*.rs']
		}),

		// In dev mode, call `npm run start` once
		// the bundle has been generated
		!production && serve(),

		// Watch the `public` directory and refresh the
		// browser on changes when not in production
		!production && livereload('public'),

		// If we're building for production (npm run build
		// instead of npm run dev), minify
		production && terser()
	],
	watch: {
		clearScreen: false
	}
};
```

**Comment by Pauan** [comment]
Thanks! Hmmm, I don't see anything wrong in there, how are you importing the `Cargo.toml` files?

**Comment by jacobmischka** [comment]
I'm doing it from within Svelte component script tags, maybe that's why?

```svelte
<script>
    import utils from '../../../wasm_wrappers/attendance_report/Cargo.toml';
</script>
```

**Comment by Pauan** [comment]
Yes, that's most likely the reason. Rollup has a well defined order for the hooks, and so the only way a race condition could happen is if you're importing multiple different `Cargo.toml` files simultaneously.

Does it work if you import it outside of the `<script>` tag (in a regular `.js` file)?

**Comment by jacobmischka** [comment]
No, it still only adds it to one of them.

```js
export { default as attendanceReport } from '../../wasm_wrappers/attendance_report/Cargo.toml';
export { default as facultyCompliance } from '../../wasm_wrappers/faculty_compliance/Cargo.toml';
export { default as studentCompetencyStats } from '../../wasm_wrappers/student_competency_stats/Cargo.toml';
```

I'm importing these from the svelte components now, so maybe that's still causing it, but I need these accessible there, so I'm not sure how else to do it.

I didn't notice until now, but the generated import is actually differently structured for the correct one too for some reason?

bundle.js:

```
...
{let{importHook:e}=t,n=new URL("assets/attendance_report_wasm_wrapper-e11191ac.wasm",document.currentScript&&document.currentScript.src||document.baseURI).href;return null!=e&&(n=e(n)),await at(n),ct}
...
{let{importHook:e}=t,n="/build/assets/faculty_compliance_wasm_wrapper-212fcd79.wasm";return null!=e&&(n=e(n)),await St(n),Et}
...
{let{importHook:e}=t,n=new URL("assets/student_competency_stats_wasm_wrapper-2de2952b.wasm",document.currentScript&&document.currentScript.src||document.baseURI).href;return null!=e&&(n=e(n)),await Nt(n),Pt}
...
```

It may be worth noting that my hack still resolves it for me with the new structure.

**Comment by Pauan** [comment]
Okay, so you are importing multiple `Cargo.toml`s simultaneously, so there is a race condition here. I'll fix it.

**Comment by Pauan** [comment]
I just published version 1.0.6 which fixes this, thanks for the report.

**Comment by jacobmischka** [comment]
Thank you for the quick response!
